pyro/optim/dct_adam.py

```diff
@@ -33,6 +31,20 @@ def _transform_inverse(x, dim, duration):
     return idct(x[right], dim)[left].add_(x[left])
 
 
+def _get_mask(x, indices):
+    # create a bollean mask from subsample indices
+    mask = x.new_zeros(x.shape, dtype=torch.long, device=x.device)
+    if len(indices) > 0:
+        size = x.dim()
+        idx = []
+        for dim in range(size):
```

Comment:
nit: `for dim in range(-x.dim(), 0):` would avoid the need for `neg_dim`

Reply:
Thanks!